lib/provider/downloadall_state.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:flutter/material.dart';
+
+//This provider switches between different download states to show on settings page
+class DownloadAllState with ChangeNotifier {
+  int _downloadState = 0;
```

Comment:
make this to an enum: {initial, downloading, completed}

Reply:
Is this possible together with `ChangeNotifier` ?